libgraphics/src/path.cpp

```diff
@@ -131,12 +133,19 @@ void MCGPathCopy(MCGPathRef self, MCGPathRef& r_new_path)
 			t_new_path -> is_mutable = false;
 			r_new_path = t_new_path;
 		}
+		else
+		{
+			MCGPathDestroy(t_new_path);
```

Comment:
In this case, these functions only call MCGPathDestroy() at a single point in conjunction with setting the return value to nil so there's not much to be gained from an autopointer here.